src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -127,19 +133,21 @@ public WebAuthnRpService(RelyingParty rp,
                 } else {
                     newCred.setTransports("");
                 }
-                newCred.setParentAccount(account);
+                newCred.setParentAccountId(account.getId());
 
-                previousCredentials.add(newCred);
                 webAuthnCredentialsRepository.save(newCred);
-                account.setCredentials(previousCredentials);
-                webAuthnUserAccountRepository.save(account);
                 activeRegistrations.remove(key);
-                return Optional.of(username);
+                if (username != null) {
```

Comment:
you can move this check after line 111, since username is final you can avoid doing all the operations on an invalid condition. Check if is empty and throw the exception.
Also, check for null OR empty strings, via StringUtils.hasText()

Reply:
Done in f542717b24f7fafb14ba7bda57cde382ccbf0e4a